compiler/luci/import/include/luci/Import/SpecialNodeBuilder.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __LUCI_IMPORT_SPECIAL_NODE_BUILDER_H__
+#define __LUCI_IMPORT_SPECIAL_NODE_BUILDER_H__
+
+#include "GraphBuilderContext.h"
+#include "GraphBuilderBase.h"
+
+#include <mio/circle/schema_generated.h>
+
+namespace luci
+{
+
+/**
+ * @brief Tensor types which requires separated node
+ */
+enum class SpecialNodeBuilderType
+{
+  INPUT,
+  BUFFER,
+  OUTPUT,
+  // TODO Extend this struct here if needed to add new type of SpecialNodeBuilder
+  COUNT // Count of different types
+};
+
+/**
+ * @brief Creates nodes from given Tensor and context
+ */
+class SpecialNodeBuilder
+{
+public:
+  virtual CircleNode *build(TensorIndex tensor_idx, GraphBuilderContext *context) const = 0;
+  virtual SpecialNodeBuilderType builder_type() const = 0;
+};
+
+/**
+ * @brief Placeholder for builders of tensors with different types
+ */
+template <SpecialNodeBuilderType Type> class SpecialNodeBuilderImpl : public SpecialNodeBuilder
```

Comment:
Can there be other name not using `Impl`?
This looks javaish and I'd prefer using name that express itself for it's purpose.